src/doctypes/conflicts.js

```diff
@@ -0,0 +1,146 @@
+import mergeWith from 'lodash/mergeWith'
+import mapValues from 'lodash/mapValues'
+import { schema } from './schema'
+
+export const isDocumentUpdateConflict = error => {
+  try {
+    const data = JSON.parse(error.message)
+    return (
+      data.error === 'conflict' ||
+      data.status === '409' ||
+      data.reason === 'Document update conflict.'
+    )
+  } catch (e) {
+    return false
+  }
+}
+
+const DEFAULT_CONFLICT_RESOLUTION_POLICY = 'server'
+
+/**
+ * Zips values contained in valuesByName entries. Each item of the
+ * resulting array is an object keyed by the "name" under which the
+ * value was found.
+ *
+ * @example
+ * ```
+ * > objectZip({
+ *   server: [{ id: 1, name: 'toto'}, { id: 2: name: 'foo'}],
+ *   user: [{ id: 1, name: 'tata'}]
+ * })
+ * [
+ *   {
+ *     server: { id: 1, name: 'toto' },
+ *     user: { id: 1, name: 'tata'}
+ *    },
+ *   { server: { id: 2, name: 'foo'} }
+ * ]
+ * ```
+ */
+const objectZip = (valuesByName, idAttribute) => {
+  const groups = {}
+
+  Object.entries(valuesByName).forEach(([valueName, values]) => {
+    values.forEach(value => {
+      const id = value[idAttribute]
+      groups[id] = groups[id] || {}
+      Object.assign(groups[id], {
```

Comment:
Object.assign (and not spread) because this code can be run in Node?

Reply:
Because I am changing the original object.